src/Common/src/System/Net/SafeCloseSocket.Windows.cs

```diff
@@ -33,52 +33,56 @@ public ThreadPoolBoundHandle IOCPBoundHandle
         // Binds the Socket Win32 Handle to the ThreadPool's CompletionPort.
         public ThreadPoolBoundHandle GetOrAllocateThreadPoolBoundHandle(bool trySkipCompletionPortOnSuccess)
         {
+            // Check to see if the socket native _handle is already
+            // bound to the ThreadPool's completion port.
+            if (_released || _iocpBoundHandle == null)
```

Comment:
You mean changing the condition from:
```C#
if (_released || _iocpBoundHandle == null) 
```
to instead just be:
```C#
if (_iocpBoundHandle == null) 
```
such that we don't throw the ObjectDisposedException from here but instead presumably have it throw some kind of exception when the handle is actually used?

Sounds reasonable, but I've not tried it.  Not sure if we have tests for that, either.  But it'd be worth checking on subsequently.